pilot/pkg/features/pilot.go

```diff
@@ -427,15 +427,6 @@ var (
 	EnableUnsafeAdminEndpoints = env.Register("UNSAFE_ENABLE_ADMIN_ENDPOINTS", false,
 		"If this is set to true, dangerous admin endpoints will be exposed on the debug interface. Not recommended for production.").Get()
 
-	XDSAuth = env.Register("XDS_AUTH", true,
```

Comment:
I was using this on test env ( to not bother with creds and using the plain text port ), but I think it can be replaced with an extra check on client IP ( sending JWTs on plain text is not generally allowed ). We need to talk about what we plan to do about the non-mTLS port in Istiod ( which is actually ok if the network is secured, including if Istiod is behind a ztunnel and we can get the peer from ztunnel MDS or directly looking up in the ambient index by IP)

Reply:
This doesn't impact the plaintext port afaik, that one is still open. But like you said we need to improve that.

So I am not sure why this would be used for test env - that might mean I am misunderstanding its usage?